src/ring_theory/algebraic.lean

```diff
@@ -152,6 +166,27 @@ begin
   exact is_integral_trans L_alg A_alg,
 end
 
+theorem alg_hom_bijective_of_is_algebraic
+  (ha : algebra.is_algebraic K L) (f : L →ₐ[K] L) : function.bijective f :=
+begin
+  refine ⟨f.to_ring_hom.injective, λ b, _⟩,
+  obtain ⟨p, hp, he⟩ := ha b,
+  let f' : p.root_set L → p.root_set L :=
+    set.maps_to.restrict f _ _ (root_set_maps_to (map_ne_zero hp) f),
+  have : function.surjective f' := fintype.injective_iff_surjective.1
+    (λ _ _ h, subtype.eq $ f.to_ring_hom.injective $ subtype.ext_iff.1 h),
+  obtain ⟨a, ha⟩ := this ⟨b, (mem_root_set_iff hp b).2 he⟩,
+  exact ⟨a, subtype.ext_iff.1 ha⟩,
+end
+
+/-- Bijection between algebra equivalences and algebra homomorphisms -/
+noncomputable
+def alg_equiv_equiv_alg_hom (ha : algebra.is_algebraic K L) : (L ≃ₐ[K] L) ≃ (L →ₐ[K] L) :=
```

Comment:
```suggestion
@[simps] def alg_equiv_equiv_alg_hom (ha : algebra.is_algebraic K L) : (L ≃ₐ[K] L) ≃ (L →ₐ[K] L) :=
```
Also, can you prove that this equiv respects composition (it is a monoid equiv, but `L →ₐ[K] L` doesn't seem to be a monoid)?

Reply:
I've introduced alg_hom.End by copying [alg_equiv.aut](https://leanprover-community.github.io/mathlib_docs/algebra/algebra/basic.html#alg_equiv.aut) and showed this is a `mul_equiv`. Moreover I moved the finite_dimensional case to this file but kept the original names. Let's see if CI passes!

Reply:
Alright CI passed. @riccardobrasca Can you check if you're happy with the new changes? Due to the upgrade from `equiv` to `mul_equiv` I have to insert `.to_equiv` at two occasions in galois.lean. Moreover [one proof in an apparently unrelated file broke](https://github.com/leanprover-community/mathlib/pull/15873/commits/21f861e7438aec7c2a743c54bb29e5eb76abf5c4) and I'm not sure why ...